IndoorMapsFrontend/src/components/pages/Directory.tsx

```diff
@@ -0,0 +1,63 @@
+import { Button, Group, JsonInput } from "@mantine/core";
+import React, { useEffect } from "react";
+import { graphql, usePreloadedQuery, useQueryLoader } from "react-relay";
+import type { PreloadedQuery } from 'react-relay';
+import type { DirectoryGetAllBuildingsQuery } from "./__generated__/DirectoryGetAllBuildingsQuery.graphql";
+
+type Props = {
+
+}
+const GetAllBuildings = graphql`
+    query DirectoryGetAllBuildingsQuery {
+    allBuildings {
+      id
+      title
+      description
+    }
+  }
+`
+
+const Directory = ({ }: Props) => {
+    const [
+        queryReference,
+        loadQuery,
+    ] = useQueryLoader<DirectoryGetAllBuildingsQuery>(
+        GetAllBuildings,
+    );
+
+    useEffect(() => {
+        loadQuery({});
+    }, []);
+
+    return (
+        <>
+            <React.Suspense fallback="Loading">
```

Comment:
[stretch] Suggestion, you can create a nice loading indicator component maybe with spinner or something and pass that component as param as opposed to string